src/screens/Ticketing/Purchase/Payment/CreditCard/use-terminal-state.ts

```diff
@@ -148,85 +66,50 @@ export default function useTerminalState(
               },
               scaExemption: true,
             });
-        dispatch({type: 'OFFER_RESERVED', reservation: response});
+        setReservation(response);
       } catch (err) {
         console.warn(err);
         handleAxiosError(err, 'reservation');
       }
     },
-    [offers, dispatch, handleAxiosError],
+    [offers, handleAxiosError],
   );
 
   useEffect(() => {
-    if (loadingState === 'reserving-offer') reserveOffer();
-  }, [reserveOffer, loadingState]);
-
-  const initialLoadRef = useRef<boolean>(true);
-  const redirectToPaymentCaptureRef = useRef<boolean>(false);
-  const verifyingBankIdRef = useRef<boolean>(false);
-
-  function handleInitialLoadingError(event: WebViewErrorEvent) {
-    dispatch({
-      type: 'SET_ERROR',
-      errorType: 'unknown',
-      errorContext: 'terminal-loading',
-    });
-  }
-
-  function handlePaymentCallback(event: WebViewNavigation) {
+    reserveOffer();
+  }, [reserveOffer]);
+
+  /**
+   * Get response code from query param. iOS uses 'responseCode' and Android
+   * uses 'response_code'.
+   */
+  function getResponseCode(event: WebViewNavigation) {
     const params = parseURL(event.url);
-    const iosResponseCode = params['responseCode'];
-    const androidResponseCode = params['response_code'];
-    if (iosResponseCode) {
-      if (isNetsResponseCode(iosResponseCode))
-        dispatch({
-          type: 'SET_NETS_RESPONSE_CODE',
-          responseCode: iosResponseCode,
-        });
-    } else if (androidResponseCode) {
-      if (isNetsResponseCode(androidResponseCode))
-        dispatch({
-          type: 'SET_NETS_RESPONSE_CODE',
-          responseCode: androidResponseCode,
-        });
-    }
+    return params['responseCode'] ?? params['response_code'];
   }
 
-  function handleVerifyingBankId() {
-    dispatch({type: 'VERIFYING_BANK_ID'});
-  }
+  const onWebViewLoadEnd = () => {
+    setIsLoading(false);
+  };
 
-  const onWebViewLoadEnd = (event: WebViewNavigation) => {
+  const onWebViewNavigationChange = (event: WebViewNavigation) => {
     const {url} = event;
-
-    // load events might be called several times
-    // for each type of resource, html, assets, etc
-    // so we have a "loading guard" here
-
-    if (initialLoadRef.current) {
-      initialLoadRef.current = false;
-      dispatch({type: 'TERMINAL_LOADED'});
-    } else if (!redirectToPaymentCaptureRef.current) {
-      if (
-        url.includes('/ticket/v2/payments/') ||
-        url.includes('atb://ticketing')
```

Comment:
we are not using this anymore? does it have any consequences? was it redundant?

Reply:
It was pretty complex code. My strategy was to remove as much as possible, and see if it still works 😅 It was probably not redundant with the old implementation where several loading states was tracked, but with this new implementation it should no longer be necessary.